### ISSUE TITLE
Missing WebException from list of what HttpClient.SendMessage can throw. 

### ISSUE THREAD

**Issue by ryanmajidi** [dotnet/dotnet-api-docs#2920]
**Submitted to [site feedback](https://github.com/MicrosoftDocs/feedback/issues/1793) by @SittenSpynne**

Link to article:
https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=netcore-2.2#System_Net_Http_HttpClient_SendAsync_System_Net_Http_HttpRequestMessage_System_Threading_CancellationToken_

Problem:
I read the documentation and assumed any reasonable problem this method would encounter would throw HttpRequestException. So I wrote code to handle that gracefully and proceeded to test.

In a Xamarin Forms project on iOS, if the network is unavailable, this method will throw a WebException.

All exceptions this method can throw should be listed. If this is a Xamarin bug, they should follow the documentation when implementing their stuff.

**Comment by mairaw** [comment]
Thanks for your feedback @SittenSpynne. @davidsh @karelz can you confirm this is the behavior and for which versions?

**Comment by davidsh** [comment]
>In a Xamarin Forms project on iOS, if the network is unavailable, this method will throw a WebException.

The Xamarin networking stack uses a different implementation.

@marek-safar can comment on the Xamarin implementation. Perhaps our documentation for HttpClient can mention these differences in behavior in the 'Remarks' section.

**Comment by marek-safar** [comment]
I don't think this is Xamarin specific issue. The implementation possibly calls user code via HttpMessageHandler hence this method can with current implementation throw any exception

**Comment by gewarren** [comment]
Since the method can throw any exception, I'm going to close this issue.
